File: isegoria/src/main/java/com/eulersbridge/isegoria/auth/login/EmailVerificationFragment.java
package com.eulersbridge.isegoria.auth.login;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.eulersbridge.isegoria.R;

public class EmailVerificationFragment extends Fragment {

    private EmailVerificationViewModel viewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.email_verification, container, false);

        viewModel = ViewModelProviders.of(this).get(EmailVerificationViewModel.class);

        final Button verifiedButton = rootView.findViewById(R.id.verifiedButton);
        verifiedButton.setOnClickListener(view -> {
            verifiedButton.setEnabled(false);

            viewModel.userVerified().observe(this, success -> {
                if (success != null && !success)
                    verifiedButton.setEnabled(true);
            });
        });

        final Button resendVerificationButton = rootView.findViewById(R.id.resendVerificationButton);
        resendVerificationButton.setOnClickListener(view -> {
            view.setEnabled(false);

            viewModel.resendVerification().observe(this,
                    success -> view.setEnabled(true));
        });

        return rootView;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        viewModel.onExit();
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/auth/login/EmailVerificationFragment.kt
package com.eulersbridge.isegoria.auth.login

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.eulersbridge.isegoria.R
import kotlinx.android.synthetic.main.email_verification.*

class EmailVerificationFragment : Fragment() {

    private val viewModel: EmailVerificationViewModel by lazy {
        ViewModelProviders.of(this).get(EmailVerificationViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View?
        = inflater.inflate(R.layout.email_verification, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        verifiedButton.setOnClickListener {
            it.isEnabled = false

            viewModel.userVerified().observe(this, Observer { success ->
                if (success == null || success == false)
                    verifiedButton.isEnabled = true
            })
        }

        resendVerificationButton.setOnClickListener { button ->
            button.isEnabled = false

            viewModel.resendVerification().observe(this, Observer {
                button.isEnabled = true
            })
        }
    }

    override fun onDestroy() {
        super.onDestroy()

        viewModel.onExit()
    }
}